owncloudData/src/test/java/com/owncloud/android/data/appregistry/datasources/implementation/OCLocalAppRegistryDataSourceTest.kt

```diff
@@ -0,0 +1,132 @@
+    }
+
+    @Test
+    fun `getAppRegistryForMimeTypeAsStream returns a Flow with an AppRegistryMimeType`() = runTest {
+
+        every { appRegistryDao.getAppRegistryForMimeType(OC_ACCOUNT_NAME, mimeTypeDir) } returns flowOf(OC_APP_REGISTRY_ENTITY)
+
+        val appRegistry = ocLocalAppRegistryDataSource.getAppRegistryForMimeTypeAsStream(OC_ACCOUNT_NAME, mimeTypeDir)
+
+        val result = appRegistry.first()
+        assertEquals(OC_APP_REGISTRY_MIMETYPE, result)
+
+        verify(exactly = 1) { appRegistryDao.getAppRegistryForMimeType(OC_ACCOUNT_NAME, mimeTypeDir) }
+    }
+
+    @Test
+    fun `getAppRegistryForMimeTypeAsStream returns a Flow with null when there are no app registries for that mime type`() = runTest {
+
+        every { appRegistryDao.getAppRegistryForMimeType(OC_ACCOUNT_NAME, mimeTypeDir) } returns flowOf(null)
+
+        val appRegistry = ocLocalAppRegistryDataSource.getAppRegistryForMimeTypeAsStream(OC_ACCOUNT_NAME, mimeTypeDir)
+
+        val result = appRegistry.first()
+        assertNull(result)
+
+        verify(exactly = 1) { appRegistryDao.getAppRegistryForMimeType(OC_ACCOUNT_NAME, mimeTypeDir) }
+    }
+
+    @Test
+    fun `getAppRegistryWhichAllowCreation returns a Flow with a list of AppRegistryMimeType`() = runTest {
+
+        every { appRegistryDao.getAppRegistryWhichAllowCreation(OC_ACCOUNT_NAME) } returns flowOf(listOf(OC_APP_REGISTRY_ENTITY))
+
+        val appRegistry = ocLocalAppRegistryDataSource.getAppRegistryWhichAllowCreation(OC_ACCOUNT_NAME)
+
+        val result = appRegistry.first()
+        assertEquals(listOf(OC_APP_REGISTRY_MIMETYPE), result)
+
+        verify(exactly = 1) { appRegistryDao.getAppRegistryWhichAllowCreation(OC_ACCOUNT_NAME) }
+    }
+
+    @Test
+    fun `getAppRegistryWhichAllowCreation returns an empty list when there are no app registries that allow creation`() = runTest {
+
+        every { appRegistryDao.getAppRegistryWhichAllowCreation(OC_ACCOUNT_NAME) } returns flowOf(emptyList())
+
+        val appRegistry = ocLocalAppRegistryDataSource.getAppRegistryWhichAllowCreation(OC_ACCOUNT_NAME)
+
+        val result = appRegistry.first()
+        assertEquals(emptyList<AppRegistryMimeType>(), result)
+
+        verify(exactly = 1) { appRegistryDao.getAppRegistryWhichAllowCreation(OC_ACCOUNT_NAME) }
+    }
+
+    @Test
+    fun `saveAppRegistryForAccount saves an AppRegistry correctly`() {
+
+        val appRegistry = AppRegistry(
+            OC_ACCOUNT_NAME, mutableListOf(
+                OC_APP_REGISTRY_MIMETYPE,
+                OC_APP_REGISTRY_MIMETYPE.copy(name = "appRegistryMimeTypes.name2")
+            )
+        )
+
+        ocLocalAppRegistryDataSource.saveAppRegistryForAccount(appRegistry)
+
+        verify(exactly = 1) {
+            appRegistryDao.deleteAppRegistryForAccount(appRegistry.accountName)
+            appRegistryDao.upsertAppRegistries(listOf(OC_APP_REGISTRY_ENTITY, OC_APP_REGISTRY_ENTITY.copy(name = "appRegistryMimeTypes.name2")))
```

Comment:
Could we set the string in a variable in the same test?

Reply:
Yes, it would make sense 👍